core/client/src/test/java/alluxio/client/file/policy/LocalFirstWithoutEvictionPolicyTest.java

```diff
@@ -0,0 +1,85 @@
+ */
+
+package alluxio.client.file.policy;
+
+import alluxio.CommonTestUtils;
+import alluxio.Constants;
+import alluxio.client.block.BlockWorkerInfo;
+import alluxio.util.network.NetworkAddressUtils;
+import alluxio.wire.WorkerNetAddress;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Tests {@link LocalFirstWithoutEvictionPolicy}.
+ */
+public class LocalFirstWithoutEvictionPolicyTest {
+  private static final int PORT = 1;
+
+  /**
+   * Tests that the local host is returned first.
+   */
+  @Test
+  public void getLocalFirst() {
+    String localhostName = NetworkAddressUtils.getLocalHostName();
+    LocalFirstWithoutEvictionPolicy policy = new LocalFirstWithoutEvictionPolicy();
+    List<BlockWorkerInfo> workerInfoList = new ArrayList<>();
+    workerInfoList.add(new BlockWorkerInfo(new WorkerNetAddress().setHost("worker1")
+        .setRpcPort(PORT).setDataPort(PORT).setWebPort(PORT), Constants.GB, 0));
+    workerInfoList.add(new BlockWorkerInfo(new WorkerNetAddress().setHost(localhostName)
+        .setRpcPort(PORT).setDataPort(PORT).setWebPort(PORT), Constants.GB, 0));
+    Assert.assertEquals(localhostName,
+        policy.getWorkerForNextBlock(workerInfoList, Constants.MB).getHost());
+  }
+
+  /**
+   * Tests that another worker is picked in case the local host does not have enough space.
+   */
+  @Test
+  public void getOthersWhenNotEnoughSpaceOnLocal() {
+    String localhostName = NetworkAddressUtils.getLocalHostName();
+    LocalFirstWithoutEvictionPolicy policy = new LocalFirstWithoutEvictionPolicy();
+    List<BlockWorkerInfo> workerInfoList = new ArrayList<>();
+    workerInfoList.add(new BlockWorkerInfo(new WorkerNetAddress().setHost("worker1")
+        .setRpcPort(PORT).setDataPort(PORT).setWebPort(PORT), Constants.GB, 0));
+    workerInfoList.add(new BlockWorkerInfo(new WorkerNetAddress().setHost(localhostName)
+        .setRpcPort(PORT).setDataPort(PORT).setWebPort(PORT), Constants.MB, Constants.MB));
+    Assert.assertEquals("worker1",
+        policy.getWorkerForNextBlock(workerInfoList, Constants.MB).getHost());
+  }
+
+  /**
+   * Tests that local host is picked if none of the workers has enough availability.
+   */
+  @Test
+  public void getLocalWhenNoneHasSpace() {
+    String localhostName = NetworkAddressUtils.getLocalHostName();
+    LocalFirstWithoutEvictionPolicy policy = new LocalFirstWithoutEvictionPolicy();
+    List<BlockWorkerInfo> workerInfoList = new ArrayList<>();
+    workerInfoList.add(new BlockWorkerInfo(new WorkerNetAddress().setHost("worker1")
+        .setRpcPort(PORT).setDataPort(PORT).setWebPort(PORT), Constants.GB, 0));
+    workerInfoList.add(new BlockWorkerInfo(new WorkerNetAddress().setHost(localhostName)
+        .setRpcPort(PORT).setDataPort(PORT).setWebPort(PORT), Constants.GB, Constants.MB));
+    Assert.assertEquals(localhostName,
+        policy.getWorkerForNextBlock(workerInfoList, Constants.GB).getHost());
```

Comment:
Because getAvailableBytes() is not return the actual available bytes of the worker, because the used bytes are updated only after a file commits. So, I need to reserve some space to get the available bytes. But, as you say, this policy would happened that the worker1 actually has 1GB, but the client do not get it to store 1GB data.